### DISCUSSION TITLE
Proper documentation for new dynamic theme capabilities

### DISCUSSION THREAD

**Discussion by fitterfizzle** [ant-design/ant-design#38715] [General]
### Reproduction link

[https://ant.design/](https://ant.design/)

### Steps to reproduce

Currently there is no proper documentation how to integrate the new theme capabilities

### What is expected?

A good documentation so that the toolbox of the ant homepage can be implemented.

![image](https://user-images.githubusercontent.com/96863959/202740901-1ea340c3-b627-4fc5-ac46-94980723a983.png)

### What is actually happening?

Very hard to understand the whole new concept with algorithms, seeds, tokens

| Environment | Info |
| --- | --- |
| antd | 5.0.0 |
| React | 17 |
| System | macOS |
| Browser | Chrome |

**Comment by afc163** [comment]
See https://ant.design/docs/react/customize-theme-v5

**Reply by fitterfizzle** [reply]
I have already tried using that but there is nowhere explained how to achieve what I need. How can I dynamically create a switcher from let's say dark to light theme? How can I get the info which current theme is being used?

**Reply by av-k** [reply]
The page is broken, please fix the links from [https://ant.design/docs/react/customize-theme-v5](https://ant.design/docs/react/customize-theme-v5) to [https://ant.design/docs/react/customize-theme](https://ant.design/docs/react/customize-theme) in your documentation

**Comment by fitterfizzle** [comment]
I have already tried using that but there is nowhere explained how to achieve what I need. How can I dynamically create a switcher from let's say dark to light theme? How can I get the info which current theme is being used?

**Reply by ashutosh887** [reply]
Yeah same. I want to customize both light and dark theme
How to do that?

**Reply by arvinxx** [reply]
@fitterfizzle  @ashutosh887 here: https://ant-design.github.io/antd-style/guide/switch-theme#4-%E7%BB%93%E5%90%88-theme-%E9%85%8D%E7%BD%AE%E8%BF%9B%E8%A1%8C%E4%B8%BB%E9%A2%98%E8%87%AA%E5%AE%9A%E4%B9%89

**Comment by afc163** [comment]
@MadCcc

**Comment by fitterfizzle** [comment]
Would something like this be the correct way to do things?

```js
theme = {
    {
        algorithm: [
            isDarkTheme ? theme.darkAlgorithm : theme.defaultAlgorithm,
            theme.compactAlgorithm,
        ],
    }
}
```

where `isDarkTheme` is a `localStorage` variable?

**Reply by nwekarsharif** [reply]
@fitterfizzle did you find any solution?

**Reply by fitterfizzle** [reply]
Not really no answer neither from @afc163 nor @MadCcc

**Reply by ashutosh887** [reply]
@fitterfizzle How to create custom theme for dark mode and light mode and change them dynamically?

**Comment by abenhamdine** [comment]
see https://ant-design.github.io/antd-token-previewer/

**Comment by X-lem** [comment]
I'd also like to see documentation for this. The slider on how to change radius, and also want to know how to switch between light/dark theme.

**Comment by fitterfizzle** [comment]
Any news @afc163 and @MadCcc ?

**Comment by rooneyhoi** [comment]
hi everyone, any updates pls?

**Reply by arvinxx** [reply]
We are working on an application level solution based on antd v5 token system. here is some demos: 

https://alpha.antd-style.arvinx.app/guide/switch-theme

And now is only Chinese version，you can read it with Translator.Also you can try it with the demos.

**Reply by rooneyhoi** [reply]
Thanks for your response, it looks promissing. I also found [ant-design cssinjs](https://ant-design.github.io/cssinjs/), is it the same approach pls?

**Reply by arvinxx** [reply]
> Thanks for your response, it looks promissing. I also found [ant-design cssinjs](https://ant-design.github.io/cssinjs/), is it the same approach pls?

They are different. `@ant-design/cssinjs` is a library for component level css-in-js solution, with has lots of optimization for antd component development. It's too verbose for application level to use. `antd-style` will be an application-level css-in-js solution based on antd v5 token system and emotion.

**Comment by dmytro-machulskyi** [comment]
is there any chance to get an answer about how to provide own design for tabs in v5?

**Reply by rooneyhoi** [reply]
I believe you can overwrite the Tabs component by using your own css tyle to overwrite it such as
`<Tabs style={{...your style here}} />`

Either or, you can use ConfigProvider to create custom theme at component level

**Reply by dmytro-machulskyi** [reply]
What I was used to implement custom Tab buttons is `renderTabBar` which takes JSX.Element for rendering. But using this approach I have to check `active` state and provide my own `onClick` function. But it looks like a good solution anyway. 
Using theme overrides not covering such cases like remove bottom border, for example. But can be useful for providing your design system colors, box-shadows, etc.

**Reply by arvinxx** [reply]
> What I was used to implement custom Tab buttons is `renderTabBar` which takes JSX.Element for rendering. But using this approach I have to check `active` state and provide my own `onClick` function. But it looks like a good solution anyway. Using theme overrides not covering such cases like remove bottom border, for example. But can be useful for providing your design system colors, box-shadows, etc.

Maybe you can wait for this RFC: https://github.com/ant-design/ant-design/discussions/40221
